builtin/mainmenu/textures.lua

```diff
@@ -145,6 +145,17 @@ function mm_texture.set_game(identifier,gamedetails)
 	
 	local path = gamedetails.path .. DIR_DELIM .."menu" ..
 									 DIR_DELIM .. identifier .. ".png"
+	if core.setting_getbool("random_menu_images") == true then
+		local count = gamedetails.image_count
+		if tonumber(count) == nil then
+			count = 1
+		end
+		local new_path = path .. "." .. math.random(1, count)
+		if io.open(new_path,"r") ~= nil and math.random(1, (count+1)) ~= 1 then
```

Comment:
Instead of requiring a setting with the number of images you could also just try to use up to 9 images.